src/OctreeDistributedDiscreteModels.jl

```diff
@@ -1,6 +1,6 @@
 
 
-struct OctreeDistributedDiscreteModel{Dc,Dp,A,B,C,D,E} <: GridapDistributed.AbstractDistributedDiscreteModel{Dc,Dp}
+mutable struct OctreeDistributedDiscreteModel{Dc,Dp,A,B,C,D,E} <: GridapDistributed.AbstractDistributedDiscreteModel{Dc,Dp}
```

Comment:
why do we need `mutable` for this type?

Reply:
Yeah I know... It's quite unfortunate, but you can only attach finalizers to mutable types... See [this](https://discourse.julialang.org/t/finalizer-only-works-with-mutable-structs/28777) for reference.

Reply:
Thanks for the reference. I did not know (or remembered) this.